public-dashboard-client/src/viz-recidivism-single-followup/VizRecidivismSingleFollowup.test.js

```diff
@@ -0,0 +1,227 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+
+import React from "react";
+import {
+  DEMOGRAPHIC_UNKNOWN,
+  DIMENSION_DATA_KEYS,
+  DIMENSION_KEYS,
+  DIMENSION_MAPPINGS,
+  TOTAL_KEY,
+} from "../constants";
+import { getDataFixture, render, screen, within } from "../testUtils";
+import { typecastRecidivismData } from "../utils";
+import VizRecidivismSingleFollowup from ".";
+
+const recidivismRates = getDataFixture(
+  "recidivism_rates_by_cohort_by_year.json"
+).map(typecastRecidivismData);
+
+function getBarLabelData(records) {
+  return records.map((record) => ({
+    label: record.releaseCohort,
+    pct: Math.round(record.recidivismRate * 100),
+  }));
+}
+test("does not explode", () => {
+  render(
+    <VizRecidivismSingleFollowup
+      data={{ followupYears: 3, recidivismRates }}
+      dimension="total"
+    />
+  );
+  expect(screen.getByText("Total")).toBeVisible();
+});
+
+test("renders a single chart for totals", () => {
+  render(
+    <VizRecidivismSingleFollowup
+      data={{ followupYears: 3, recidivismRates }}
+      dimension="total"
+    />
+  );
+  // our chart library generates text labels for all its visual marks;
+  // we can use this as a proxy for proper visual rendering
+  const vizEl = screen.getByLabelText("8 bars in a bar chart");
```

Comment:
that's so cool! This may be early to think about but I wonder if we would want to prepare for additional years of data, and if we should get the number `8` in this string (and the others) from from the test fixtures?

Reply:
I don't think that there is actually a plan for there to be additional years of data? the years may change but I think it will remain as a moving ten-year window. in which case the conditions of these tests would not change regardless of the years represented. 

if that were to change (or if I'm just wrong) then I'd agree we should do what you say